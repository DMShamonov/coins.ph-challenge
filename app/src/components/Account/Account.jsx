// @flow

import * as React from 'react';

import User from 'components/User';
import Button from 'components/Button';
import Icon from 'components/Icon';

import type { PropsType } from './types.js.flow';

export default function Account({
  name,
  avatar = '',
  onLogout,
}: PropsType): React.Node {
  return (
    <div className="account">
      <User avatar={avatar} name={name} className="account__user" />
      <Button theme="shell" className="account__logout" onClick={onLogout}>
        <Icon theme="logout" width={24} title="Logout" />
      </Button>
    </div>
  );
}
