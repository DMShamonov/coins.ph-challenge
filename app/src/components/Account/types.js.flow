// @flow

export type PropsType = {|
  name: string,
  avatar?: string,
  onLogout: () => void,
|};
