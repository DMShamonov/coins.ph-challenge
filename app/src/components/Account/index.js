import { connect } from 'react-redux';

import SESSION_ACTION_TYPES from 'redux/actionTypes/session';

import Account from './Account';

function mapStateToProps({ common: { profile } }) {
  const { name } = profile || {};

  return {
    name,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onLogout() {
      dispatch({ type: SESSION_ACTION_TYPES.LOGOUT });
    },
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Account);
