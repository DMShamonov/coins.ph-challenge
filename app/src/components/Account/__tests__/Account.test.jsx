import React from 'react';
import { shallow, mount } from 'enzyme';

import Account from '../Account';

describe('`Account` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Account name="__USER_NAME__" onLogout={jest.fn()} />)).toMatchSnapshot();
  });

  test('with `avatar` prop', () => {
    expect(shallow(<Account name="__USER_NAME__" avatar="__AVATAR__" onLogout={jest.fn()} />)).toMatchSnapshot();
  });
});

describe('Check `Account` props', () => {
  describe('Check callback props', () => {
    test('`onLogout` called', () => {
      const onLogout = jest.fn();
      const wrapper = mount(<Account name="__USER_NAME__" onLogout={onLogout} />);

      wrapper.find('button').simulate('click');

      expect(onLogout).toHaveBeenCalled();
    });
  });
});
