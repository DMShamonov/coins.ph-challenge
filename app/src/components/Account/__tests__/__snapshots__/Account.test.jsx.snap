// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Account\` component should render correctly with \`avatar\` prop 1`] = `
<div
  className="account"
>
  <User
    avatar="__AVATAR__"
    className="account__user"
    name="__USER_NAME__"
  />
  <Button
    className="account__logout"
    onClick={[MockFunction]}
    theme="shell"
  >
    <Icon
      theme="logout"
      title="Logout"
      width={24}
    />
  </Button>
</div>
`;

exports[`\`Account\` component should render correctly with default props 1`] = `
<div
  className="account"
>
  <User
    avatar=""
    className="account__user"
    name="__USER_NAME__"
  />
  <Button
    className="account__logout"
    onClick={[MockFunction]}
    theme="shell"
  >
    <Icon
      theme="logout"
      title="Logout"
      width={24}
    />
  </Button>
</div>
`;
